File: backend/app/schemas/meeting.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict

# Base schema containing shared attributes
class MeetingBase(BaseModel):
    title: str
    description: Optional[str] = None
    created_by: str
    scheduled_time: Optional[datetime] = None
    duration: Optional[int] = None  # Duration in minutes
    is_instant: bool = False

# Schema for creating a meeting (inputs)
class MeetingCreate(MeetingBase):
    pass

# Schema for returning a meeting (outputs)
class MeetingResponse(MeetingBase):
    id: int
    meeting_id: str
    created_at: datetime

    # Enable ORM compatibility for automatic conversion from SQLAlchemy models
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/meeting.ts
export interface Meeting {
  id: number;
  meeting_id: string;
  title: string;
  description?: string | null;
  created_by: string;
  scheduled_time?: string | null;
  duration?: number | null;
  created_at: string;
  is_instant: boolean;
}

export interface MeetingCreateInput {
  title: string;
  description?: string;
  created_by: string;
  scheduled_time?: string;
  duration?: number;
  is_instant?: boolean;
}

export interface InstantMeetingCreateInput {
  title?: string;
  created_by?: string;
}

export interface RecentMeeting {
  id: number;
  meeting_id: string;
  participant_name: string;
  joined_at: string;
}

export interface RecentMeetingCreateInput {
  meeting_id: string;
  participant_name: string;
}
